guides/Scripting_Languages.md

```diff
@@ -0,0 +1,320 @@
+        corresponding one's number in the source file.
+        A proper stack-trace with error lines rewritten requires the implementation of an
+        error handler function making use of the annotations. 
+    * virtual file-system
+        works out of the box.
+
+2. [Just-in-time compilation](https://en.wikipedia.org/wiki/Just-in-time_compilation)
+    * more complex setup  
+        This solution needs more additional coding inside the project. 
+    * additional dependencies required  
+        In this case the project must ship with the compiler library and its dependencies.
+    * runtime error line translation  
+        is usually done by the compiler without extra care.
+    * virtual file-system
+        The Lua loader for the language needs to make use of lovr.filesystem.read or it can't
+        access files via the virtual file-system.
+        Without a solution there are two issues:      
+        1. save directory  
+            Modules can't be required from within the LÖVR save directory.
+            This is a problem when downloading modules is supported since it is the only place
+            LÖVR allows writing to. 
+        2. fuse mode  
+            In fused mode the virtual file-system is the only working file access method.
+            Meaning the project won't work at all if it relies on 'require'.
+
+The more elegant solution is trans-compilation at runtime.
+Source and Lua files can't be out of sync and the setup is more developer friendly.
+Runtime error line translation is usually done by the compiler without extra care.
+
+The following chapters present language specific solutions for both use cases.
+Chapters are sorted in alphabetical order.
+A dialect means the language is inspired by the parent but not fully syntax compatible.
+
+Fennel (Lisp/Clojure dialect)
+---
+
+<img align="right" width="300" height="300" src="https://fennel-lang.org/logo.svg" alt="Fennel Logo">
+
+* Home: [Fennel](https://fennel-lang.org/)
+* Source: [github](https://github.com/bakpakin/Fennel)
+* Syntax: [Lisp](https://en.wikipedia.org/wiki/Lisp_\(programming_language\))/[Clojure](https://clojure.org/) inspired
+* Semantic: mostly Lua
+* jit compiler: Pure Lua
+* Dependencies: (TODO)
+
+#### ahead-of-time
+
+    $ fennel --compile yourProject
+
+(TODO: error line translation)
+
+#### just-in-time
+
+* [Setup#Embedding Fennel](https://fennel-lang.org/setup#embedding-fennel)
+* [Minimal LÖVR Setup](https://github.com/MattRoelle/min-lovr-fennel)
+
+Haxe (ECMAScript dialect)
+---
+
+<img align="right" width="200" height="200" src="https://haxe.org/img/haxe-logo.svg" alt="Haxe Logo">
+
+* Home: [Haxe](https://haxe.org/)
+* Source: [github](https://github.com/HaxeFoundation/haxe)
+* Syntax: [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) inspired
+* Semantic: (TODO)
+* jit compiler: Not available (TODO: is that true
+* Dependencies: (TODO)
+
+#### ahead-of-time
+
+[Haxe Lua](https://haxe.org/manual/target-lua.html)
```

Comment:
Here are the LÖVR haxe bindings https://github.com/excessive/lovr-haxe-wrappergen

Reply:
I am not sure where to put the URL and what to write about it.